src/test/compile-fail/associated-const-type-parameter-arrays.rs

```diff
@@ -26,7 +26,9 @@ impl Foo for Def {
 
 pub fn test<A: Foo, B: Foo>() {
     let _array: [u32; <A as Foo>::Y];
-    //~^ ERROR the trait bound `A: Foo` is not satisfied
+    //~^ ERROR cannot use an outer type parameter in this context [E0402]
```

Comment:
Yeah, I didn't really make anything worse, just better encoded the existing semantics.
We'll be able to clean this up in the future, when everything is body-based.